processors/base/helpers.py

```diff
@@ -58,10 +58,24 @@ def default(self, obj):
 
 
 def clean_dict(raw_dict):
-    """Remove False values from dict.
+    """Remove falsy values from dict.
     """
     cleaned_dict = {}
     for key, value in raw_dict.items():
         if value:
             cleaned_dict[key] = value
     return cleaned_dict
+
+
+def get_optimal_title(*strings):
+    """Return first striped string with enough length or last striped string.
+    """
+    MIN_LENGTH = 10
+    for string in strings:
+        if not string:
+            continue
+        string = string.strip()
+        string = string.strip('.')
```

Comment:
I'm not sure if this improves anything, but doing multiple strips like this might have some corner cases. For example, imagine that `str = 'foo bar.   '`, then `str.strip().strip('.')` would return `'foo bar'`, but if `str = 'foo bar  .'`, then the same code would return `'foo bar '`.

If this might be a problem, you can avoid this using: `str.strip('.' + string.whitespace)`. The `string.whitespace` is defined as `'\t\n\x0b\x0c\r '`, so you would strip all whitespace and all dots in a single pass, which always work.

This might be better or not, depends on what you expect to be in the string.